### PR TITLE
[dev-v2.9] Forward ports prometheus-federator 103.0.1+up0.4.0

### PR DESCRIPTION
**Pull request by lucasmlp**
This PR forward-ports the following `v2.8.3` related charts from `release-v2.8`:

- prometheus-federator 103.0.1+up0.4.0

### CODE REVIEW

File: index.yaml [hunk 1]
```diff
@@ -5970,27 +5970,27 @@ entries:
...
     description: Prometheus Federator
-    digest: b78bd24740ed6de630e8db8366aada8928cb224eb580ee843b099a056ab33059
+    digest: 97740e89c6f45f34c08a6d9642b94cf80a49bc6cc402b57591e07a84fadc6fb5
     icon: https://raw.githubusercontent.com/rancher/prometheus-federator/main/assets/logos/prometheus-federator.svg
     name: prometheus-federator
     urls:
-    - assets/prometheus-federator/prometheus-federator-103.0.1+up0.4.1.tgz
-    version: 103.0.1+up0.4.1
+    - assets/prometheus-federator/prometheus-federator-103.0.1+up0.4.0.tgz
+    version: 103.0.1+up0.4.0
```
Comment (adamkpickering) [inline index.yaml:6134]: This change looks suspect to me. Same thing for the lines above it in this file. Changes of this pattern are not present in any of the other forward port PRs I have seen. Are these changes supposed to be made?

File: release.yaml [hunk 2]
```diff
@@ -105,10 +106,12 @@ rancher-monitoring:
   - 104.0.0-rc1+up45.31.1
   - 103.0.3+up45.31.1
   - 102.0.3+up40.1.2
+  - 103.0.4+up45.31.1
 rancher-monitoring-crd:
   - 104.0.0-rc1+up45.31.1
   - 102.0.3+up40.1.2
   - 103.0.3+up45.31.1
+  - 103.0.4+up45.31.1
```
Comment (adamkpickering) [inline release.yaml]: I don't think we should be making changes to `rancher-monitoring` and `rancher-monitoring-crd` in this PR...

**Review by adamkpickering — CHANGES_REQUESTED** [review]
I think this PR may have some errors. Please confirm that everything is correct.

File: index.yaml (hunk 1 above)
Comment (lucasmlp) [reply index.yaml:6134]: Agreed. Manually added the entry that was modified. Now, this only adds the entry for the new asset. Thanks for catching it

File: release.yaml (hunk 2 above)
Comment (lucasmlp) [reply release.yaml]: Just rebased the branch and it's fixed.

**Review by adamkpickering — APPROVED** [review]

**Pull request merged**